packages/alfa-dom/src/node/document.ts

```diff
@@ -44,6 +49,14 @@ export class Document extends Node {
     return "/";
   }
 
+  protected _structurallyEquals(value: unknown): value is this {
+    return (
+      value instanceof Document &&
+      super._structurallyEquals(value) &&
+      Iterable.equals(this.style, value.style)
```

Comment:
We also need to implement structural equality for the CSS structures as documents with style sheets otherwise won't be considered structurally equal unless `value === this`.